### DISCUSSION TITLE
How to updated an already added custom field to Users collection

### DISCUSSION THREAD

**Discussion by talha0324** [pocketbase/pocketbase#4795] [Q&A] [answered]
Hi, 
I am using migration to extend `users` collection to add a field `is_deleted`, I set it as required initially but now I want to update it to not required.
I used a migration file for that purpose.
I updated the migration to reflect but it still says `No new migrations to apply.`
This is my migration file:

```
func init() {
	m.Register(func(db dbx.Builder) error {
		dao := daos.New(db)
		collection, err := dao.FindCollectionByNameOrId("users")
		if err != nil {
			return err
		}
		collection.Schema.AddField(&schema.SchemaField{
			Name:     "is_deleted",
			Type:     schema.FieldTypeBool,
			Required: false,
		})
		if err := dao.SaveCollection(collection); err != nil {
			return err
		}
		return nil
	}, func(db dbx.Builder) error {
		// add down queries...

		return nil
	})
}

```

**Comment by ganigeorgiev** [comment, marked as answer]
You can either run `migrate down` to rollback the last migration or create a new one that will change the field option, something like:

```go
m.Register(func(db dbx.Builder) error {
    dao := daos.New(db)

    collection, err := dao.FindCollectionByNameOrId("users")
    if err != nil {
        return err
    }

    // field is a pointer so modifying it will directly change the one from the collection.Schema
    field := collection.Schema.GetFieldByName("is_deleted")
    if field != nil {
        field.Required = false
    }

    return dao.SaveCollection(collection)
}, func(db dbx.Builder) error {
    // add down queries...

    return nil
})
```

**Reply by talha0324** [reply]
@ganigeorgiev thank you for your prompt response.
Is there a way to do up/down on specific migration?
Thanks!

**Reply by ganigeorgiev** [reply]
No. An alternative if you have executed the migration only on the local environment could be to rename the migration file and run `migrate history-sync` (_this will delete the old migration file from the `_migrations` table_) and then run `migrate up`/`serve` to execute the "new" migration.

You can run `./pocketbase migrate --help` to explore all available migrate commands.

**Reply by talha0324** [reply]
All good, thanks!
